### PR TITLE
Fix title and subtitle font styling issue with mergeOptions

### LINKED ISSUE #6597: fontWeight / fontStyle default options are not applied
### Issue Description

With the recent addition of fontWeight / fontStyle to android (https://github.com/wix/react-native-navigation/pull/6590), I missed to include those in the default options. So they are not being merged in.

### Steps to Reproduce / Code Snippets / Screenshots

```ts
Navigation.setDefaultOptions({
  topBar: {
    title: {
      fontFamily: "Cool Font",
      fontStyle: "italic",
      fontWeight: "bold",
    },
  },
});
```
---
### Environment
* React Native Navigation version: 7.0.0-snapshot.1216
* React Native version: 0.63.2
* Platform(s) (iOS, Android, or both?): both
* Device info (Simulator/Device? OS version? Debug/Release?): all

### PR DESCRIPTION
**Pull request by guyca**
When a screen is pushed into a stack without a title/subtitle and the title/subtitle are then updated later on with a call to mergeOptions - font styles like color, size and typeface were not applied.

This commit ensures that resolved font styles are always applied when updating title/subtitle with mergeOptions.
Fixes #6597

### CODE REVIEW

File: lib/android/app/src/main/java/com/reactnativenavigation/options/FontOptions.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ import android.graphics.Typeface
 import com.reactnativenavigation.options.params.NullText
 import com.reactnativenavigation.options.params.Text
 import com.reactnativenavigation.options.parsers.TypefaceLoader
+import java.lang.RuntimeException
```
Comment (danilobuerger) [inline lib/android/app/src/main/java/com/reactnativenavigation/options/FontOptions.kt:7]: Why a `java.lang` import?

**Review by danilobuerger — COMMENTED** [review]
Tested with playground, seems to work.

File: lib/android/app/src/main/java/com/reactnativenavigation/options/FontOptions.kt (hunk 1 above)
Comment (guyca) [reply lib/android/app/src/main/java/com/reactnativenavigation/options/FontOptions.kt]: W00ps 🙈

**Review by danilobuerger — APPROVED** [review]
lgtm, tested in playground.

**Pull request merged**